generics/services/knowledge-platform.js

```diff
@@ -0,0 +1,210 @@
+
+const request = require('request');
+const CREATION_PORTAL_URL = process.env.CREATION_PORTAL_URL
+
+const headers = {
+  "content-type": "application/json",
+  "Authorization": process.env.CREATION_PORTAL_AUTHORIZATION
+}
+
+const copyQuestionSet = function (copyReq, questionSetId) {
+  const copyQuestionSetUrl = CREATION_PORTAL_URL + messageConstants.endpoints.COPY_QUESTION_SET + "/" + questionSetId;
+  const options = {
+    headers,
+    json: { request: { questionset: copyReq } }
+  };
+
+
+  return new Promise((resolve, reject) => {
+    try {
+
+      const copyQuestionSetCallback = function (err, data) {
+        if (err || data.statusCode != 200) {
+          return reject({
+            message: messageConstants.apiResponses.QUESTIONSET_NOT_FOUND,
+            status: httpStatusCode.bad_request.status,
+          })
+        } else if (data.statusCode == 200) {
+          let response = data.body
+          return resolve(response);
+
+        }
+      }
+      request.post(copyQuestionSetUrl, options, copyQuestionSetCallback)
+
+    } catch (error) {
+      return reject(error);
+    }
+  })
+
+}
+
```

Comment:
@swetha-aggidevara please add comments and function documentation to all functions in this file